core/src/main/java/jenkins/model/ArtifactManager.java

```diff
@@ -60,11 +60,17 @@
      * @param launcher a launcher to use if external processes need to be forked
      * @param listener a way to print messages about progress or problems
      * @param artifacts map from paths in the archive area to paths relative to {@code workspace} (all paths {@code /}-separated)
+     * @param preserveExecute if true, sets the execute bit on the target if the execute bit is set on the source
+     * @param preserveMtime if true, sets the modification time of the target to match the modification time of the source
      * @throws IOException if transfer or copying failed in any way
      * @throws InterruptedException if transfer was interrupted
      * @see ArtifactArchiver#perform(Run, FilePath, Launcher, TaskListener)
      */
-    public abstract void archive(FilePath workspace, Launcher launcher, BuildListener listener, Map<String,String> artifacts) throws IOException, InterruptedException;
+    public abstract void archive(FilePath workspace, Launcher launcher, BuildListener listener,
```

Comment:
Clients are fine, external `ArtifactManager` implementations will break. Try deploying and use compress-artifacts-plugin:
```
java.lang.AbstractMethodError: jenkins.model.ArtifactManager.archive(Lhudson/FilePath;Lhudson/Launcher;Lhudson/model/BuildListener;Ljava/util/Map;ZZ)V
 at hudson.tasks.ArtifactArchiver.perform(ArtifactArchiver.java:238)
 at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:74)
 at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
 at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:761)
 at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:721)
 at hudson.model.Build$BuildExecution.post2(Build.java:183)
 at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:670)
 at hudson.model.Run.execute(Run.java:1775)
 at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
 at hudson.model.ResourceController.execute(ResourceController.java:89)
 at hudson.model.Executor.run(Executor.java:240)
```

But as I said, this probably need not be configurable at all.